src/main/java/com/lambdaworks/redis/SimpleRedisConnection.java

```diff
@@ -0,0 +1,142 @@
+package com.lambdaworks.redis;
+
+import com.lambdaworks.redis.SimpleSerializers.JavaSerializer;
+import com.lambdaworks.redis.codec.ByteArrayCodec;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.Serializable;
+
+/**
+ * A simple redis connection for caching and retrieval from redis
+ *
+ * @author <a href="mailto:a.abdelfatah@live.com">Ahmed Kamal</a>
+ * @since 12.01.16 09:17
+ */
+public class SimpleRedisConnection {
+
+    private final RedisClient redisClient;
+    protected static final InternalLogger logger = InternalLoggerFactory.getInstance(SimpleRedisConnection.class);
+    private JavaSerializer codec;
+
+    /**
+     * creates a simple redis connection providing redis client
+     * @param redisClient
+     */
+    public SimpleRedisConnection(RedisClient redisClient) {
+
+        if (redisClient == null)
+            throw new IllegalArgumentException("Redis Client argument shouldn't be null");
+
+        this.redisClient = redisClient;
+        this.codec = new JavaSerializer();
+    }
+
+    /**
+     * Cache generic serializable value in redis
+     * <br/>
+     * Sets field in the hash stored at key to value. If key does not exist, a new key holding a hash is created.
+     * <br/>
+     * If field already exists in the hash, it is overwritten.
+     *
+     * @param key
+     * @param field
+     * @param value that implements serializable
+     * @param <T>
+     * @return true if field is a new field in the hash and value was set.
+     * <br/> false if field already exists in the hash and the value was updated.
+     */
+    public <T extends Serializable> boolean cache(String key, String field, T value) throws Exception {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
```

Comment:
About your comment of sync calls: lettuce works internally fully async. The sync API makes just lettuce look like sync. A lettuce connection is thread-safe and can be used by multiple callers at a time.